modules/environment.nix

```diff
@@ -11,6 +11,26 @@ with lib;
 let
   cfg = config.environment;
   profileScript = pkgs.writeShellScript "profile" cfg.shell.profile;
+
+  requiredPackages = map (pkg: setPrio ((pkg.meta.priority or 5) + 3) pkg) [
+    pkgs.bashInteractive
+    pkgs.busybox
+  ];
+
+  makeShellWrapper = pkg: name:
+    let
+      script = pkgs.writeShellScriptBin name ''
+        export PATH="$PATH"':${makeBinPath cfg.systemPackages}'
+        exec ${pkg}/bin/${name} "$@"
+      '';
+    in "${script}/bin/${name}";
+
+  shells = if cfg.shell.enable then ({
+    bash = (makeShellWrapper pkgs.bashInteractive "bash");
+    sh = (makeShellWrapper pkgs.bashInteractive "bash");
+  }) else ({
+    sh = "${pkgs.busybox}/bin/sh";
+  });
```

Comment:
In this else branch i would include nothing, so its possible to completely replace `/bin/sh` with another module

Reply:
I just pushed a commit to do remove the shells when the option is not set.